File: backend/app/schemas/briefing.py
from pydantic import BaseModel


class BriefingRequest(BaseModel):
    prompt_key: str | None = None  # from guided prompts
    custom_query: str | None = None  # free-form
    user_type: str = "retail_investor"
    language: str = "en"
    perspective: dict | None = None


class BriefingResponse(BaseModel):
    prompt_used: str
    response_text: str
    sources: list[dict] = []
    follow_up_prompts: list[str] = []

File: frontend/src/types/briefing.ts
export interface GuidedPrompt {
  key: string;
  label: string;
}

export interface BriefingRequest {
  prompt_key?: string;
  custom_query?: string;
  user_type?: string;
  language?: string;
  perspective?: Record<string, number>;
}

export interface BriefingResponse {
  prompt_used: string;
  response_text: string;
  sources: Record<string, unknown>[];
  follow_up_prompts: string[];
}
